### PR TITLE
feat: `Media` `start` & `end`

### LINKED ISSUE #228: [Question] How to play video with start time?
Is there any way to play video begin from a specific time like this command:
`mpv  --start=100 loadfile 'http://host/a.mp4' append  `

### PR DESCRIPTION
**Pull request by alexmercerind**
The API is fairly easy to use; implemented for both `NativePlayer` & `WebPlayer` backends.

```dart
await player.open(
  Media(
    'https://www.example.com/video.mp4',
    start: const Duration(seconds: 5),
  ),
);
await player.open(
  Media(
    'https://www.example.com/video.mp4',
    start: const Duration(seconds: 5),
    end: const Duration(minutes: 1),
  ),
);
```
 
<hr>

Closes #228.

### CODE REVIEW

File: media_kit/lib/src/player/native/player/real.dart [hunk 1]
```diff
@@ -706,6 +712,19 @@ class NativePlayer extends PlatformPlayer {
       await waitForPlayerInitialization;
       await waitForVideoControllerInitializationIfAttached;
 
+      // External List<Media>:
+      // ---------------------------------------------
+      final map = SplayTreeMap<double, Media>.from(
```
Comment (alexmercerind) [inline media_kit/lib/src/player/native/player/real.dart:717]: I hope this `move` matches internal libmpv implementation (same is used on web; though passes unit-tests).

File: media_kit_video/lib/src/video_controller/web_video_controller/real.dart [hunk 2]
```diff
@@ -8,7 +8,7 @@ import 'dart:async';
 import 'dart:js' as js;
 import 'dart:html' as html;
 
-import 'dart_ui/dart_ui.dart';
+import 'dart:ui_web';
```
Comment (alexmercerind) [inline media_kit_video/lib/src/video_controller/web_video_controller/real.dart:11]: https://github.com/flutter/engine/pull/41877

File: media_kit/lib/src/player/native/player/real.dart (hunk 1 above)
Comment (birros) [reply media_kit/lib/src/player/native/player/real.dart:717]: I didn't know `SplayTreeMap` and the `map[to - 0.5]` trick is very elegant. Bravo!

**Review by birros — APPROVED** [review]
LGTM!

**Comment by AALAMILLO** [comment]
Hi! With the lastest version this feature doesn't appears 
image

I'm declaring this un pubspec
  media_kit: ^1.1.10                             # Primary package.
  media_kit_video: ^1.2.4                        # For video rendering.
  media_kit_libs_video: ^1.0.4                   # Native video dependencies.

**Comment by Prince-of-death** [comment]
> Hi! With the lastest version this feature doesn't appears image
> 
> I'm declaring this un pubspec media_kit: ^1.1.10 # Primary package. media_kit_video: ^1.2.4 # For video rendering. media_kit_libs_video: ^1.0.4 # Native video dependencies.

I am also experiencing this

**Pull request merged**